.vitepress/theme/components/Feedback.vue

```diff
@@ -0,0 +1,208 @@
+// We send a form to Netlify: https://docs.netlify.com/forms/setup/
+function submitForm() {
+  isSending.value = true;
+  fetch('/', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+    body: encode({
+      'form-name': 'doc-feedback',
+      page: route.path,
+      wasHelpful: currentState.value,
+      feedback: feedback.value,
+    }),
+  })
+    .then(() => {
+      currentState.value = FeedbackState.END;
+      feedback.value = '';
+      hasError.value = false;
+    })
+    .catch((error) => {
+      hasError.value = true;
+    })
+    .finally(() => {
+      isSending.value = false;
+    });
+}
+
+// Reset the states on page change
+watch(
+  () => route.path,
+  () => {
+    currentState.value = FeedbackState.START;
+    feedback.value = '';
+    hasError.value = false;
+  }
+);
+</script>
+
+<template>
+  <div class="feedbackContainer">
+    <template v-if="currentState === FeedbackState.START">
+      <p class="title">Was this page helpful?</p>
+      <div class="buttonContainer">
+        <button class="button" @click="onButtonClick(FeedbackState.YES)">
+          <span class="icon thumbsUp"></span> Yes
+        </button>
+        <button class="button" @click="onButtonClick(FeedbackState.NO)">
+          <src class="icon thumbsDown"></src>No
+        </button>
+      </div>
+    </template>
+
+    <template v-else-if="[FeedbackState.YES, FeedbackState.NO].includes(currentState)">
+      <form
+        class="form"
+        name="doc-feedback"
+        method="post"
+        @submit.prevent="submitForm"
+        data-netlify="true"
+      >
+        <input type="hidden" name="form-name" value="doc-feedback" />
+        <label class="title label" for="feedback">
+          <template v-if="currentState === FeedbackState.YES"> What was most helpful? </template>
+          <template v-else> What should we improve? </template>
+        </label>
+        <textarea
+          id="feedback"
+          name="feedback"
+          class="textarea"
+          v-model="feedback"
+          autofocus
+        ></textarea>
+        <button class="button" type="submit" :disabled="feedback.length === 0">
+          <LoadingIcon v-if="isSending" :size="11" />Submit
+        </button>
+        <p v-if="hasError" class="error">Sorry, something went wrong. Please try it again later.</p>
+      </form>
+    </template>
+    <template v-else-if="currentState === FeedbackState.END">
+      <p class="end">Thank you for helping improve StackBlitz documentation! &#9829;</p>
+    </template>
+  </div>
+</template>
+
+<style scoped lang="scss">
+.feedbackContainer {
+  margin-block-start: 50px;
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  align-items: center;
+  gap: 30px;
+  background-color: var(--vp-c-bg);
+}
+
+.buttonContainer {
+  display: flex;
+  gap: 16px;
+}
+
+.title {
+  font-weight: 700;
+  font-size: 18px;
+}
+
+.label {
+  display: block;
+  margin-block-end: 10px;
+}
+
+.button {
```

Comment:
❓Should the buttons color be different in the light mode? Right now it looks like this: 

![Screenshot 2022-11-25 at 16 01 29](https://user-images.githubusercontent.com/3702771/204015104-e24fbe69-88d3-4647-9d44-6d86b54261de.png)

Reply:
This looks alright, but to keep in sync with the theme we could try using those colors:

- icon color: the page's background color (`--vp-c-bg`)
- button background: use the main link color (`--vp-c-brand`)

Reply:
Hm so it will look like these? I feel a bit weird for the dark mode but I'm not a designer 🧐

<img width="440" alt="Screenshot 2022-12-02 at 20 32 36" src="https://user-images.githubusercontent.com/3702771/205371844-60368618-2727-44d9-9768-075db4227758.png">
<img width="455" alt="Screenshot 2022-12-02 at 20 33 57" src="https://user-images.githubusercontent.com/3702771/205371849-b1f7805d-fe92-48a2-b660-57185a8d7930.png">